### ISSUE TITLE
search: encode () in the url pattern properly

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/chaise#949]
I ma using a url pattern to link experiment_type to its record page. The experiment_type has space and `()`, the space is encoded, but not `()`. 

Here is the link to the search page that generate incorrect link
https://dev.gpcrconsortium.org/chaise/search/#1/experiments:experiment

Here is the annotation that I used to generate the url. 
--('experiments', 'experiment', 'experiment_type', 'comment', '["url"]')
 -- ('experiments', 'experiment', 'experiment_type','description','{"url_pattern" : "/chaise/record/#1/experiments:experiment_type/name={value}"}')

@karlcz mentioned that you probably need an optional pattern syntax to ask for values to be url-encoded.  adding it blindly may break other existing deployments if they have already encoded data in the table and are just appending together url fragments.  also, each expanded parameter needs to be encoded separately.  you cannot append the URL first and then encode later.

**Comment by svoinea** [comment]
The problem was fixed through the commit 34990a635d6abf86d1b1bc69c500bcfcde1c75f6 by allowing the usage of the additional pattern `{encode.value}` when the URI encoded value is required.

As the above mentioned annotations were not present in the database, I have added them:
```
insert into model_column_annotation(schema_name,table_name, column_name, annotation_uri,annotation_value) values('experiments','experiment','experiment_type','comment','["url"]');

insert into model_column_annotation(schema_name,table_name, column_name, annotation_uri,annotation_value) values('experiments','experiment','experiment_type','description','{"url_pattern" : "/chaise/record/#1/experiments:experiment_type/name={encode.value}"}');

```

**Comment by svoinea** [comment]
@hongsudt Ready for review.
